### DISCUSSION TITLE
Unable to Connect to PUB400 FROM VS Code for IBM i

### DISCUSSION THREAD

**Discussion by brianrocket13** [codefori/vscode-ibmi#2337] [Q&A] [answered]
I can connect to PUB400 without any issues in a TN5250 emulator session whether my Norton VPN is running or not. If I try to connect through VS Code for IBM i, the following error is displayed: Not connected to pub400.com! Check your credentials (All configured authentication methods failed).  

If I enter the following SSH command from a command line it rejects my password. My password is mixed case with special characters and numbers and works fine when I use it in a Telnet session. The PUB400.com site will not take my password either if I try to go sign in so I can go to the PUB400 Forum.    

C:\Users\brian>ssh BEBPGMR13@pub400.com -p 2222
*********************************************
* WELCOME TO PUB400.COM - your public IBM i *
* ----------------------------------------- *
* Please take note of the following:        *
* - all access is logged                    *
* - please be polite and do not disturb     *
*   other users!                            *
* - this system is of limited support       *
* - see https://pub400.com                  *
*********************************************

-> Enter your password for Logon below:
BEBPGMR13@pub400.com's password:
Permission denied, please try again.

**Comment by sebjulliand** [comment, marked as answer]
Sounds like a pure SSH issue.
For the sake of testing, could you change your password to remove any special characters and keep only letters and numbers; then try to connect?
Then feel free to switch back to your previous password. I just wonder if special characters could mess with the SSH authentication.

**Reply by brianrocket13** [reply]
I got rid of the '@' in the password and I was able to sign on in VS Code and in the tabs on the PUB400 website.  Thank you!

**Comment by chrjorgensen** [comment]
@brianrocket13 Technically this is not a Code for IBM i issue - you have an issue with SSH and PUB400, and not with Code for IBM i.
I would suggest you contact the PUB400 administrators to ask why your SSH connection fails.

Telnet and SSH are two completely different services and protocols, so even if the simple one works (telnet) there may still be problems in the complex one (SSH), e.g. username length (not an issue on PUB400), authentication settings, home directory, authority on files and directories etc. Your best bet is the PUB400 administrators... you can also use the `-vvv` flag on the SSH command to get more debug information when connecting.

**Reply by brianrocket13** [reply]
Fortunately I removed the special characters from my password and it all works fine now. Thank you for commenting.
